flink-mesos/src/main/java/org/apache/flink/mesos/scheduler/Offer.java

```diff
@@ -81,11 +82,14 @@ public Offer(Protos.Offer offer) {
 		}
 		this.resources = Collections.unmodifiableList(resources);
 
-		this.cpuCores = aggregateScalarResource(resourceMap, "cpus");
-		this.memoryMB = aggregateScalarResource(resourceMap, "mem");
-		this.networkMbps = aggregateScalarResource(resourceMap, "network");
-		this.diskMB = aggregateScalarResource(resourceMap, "disk");
-		this.portRanges = Collections.unmodifiableList(aggregateRangesResource(resourceMap, "ports"));
+		Map<String, Double> aggregatedScalarResourceMap = scalarResourceMap.entrySet()
+			.stream()
+			.collect(Collectors.toMap(
+				e -> e.getKey(),
+				e -> e.getValue().stream().mapToDouble(r -> r.getScalar().getValue()).sum()
+			));
+		this.aggregatedScalarResourceMap = Collections.unmodifiableMap(aggregatedScalarResourceMap);
```

Comment:
I think this change was motivated by a need to implement the new `getScalarValues`.  It is good that the aggregation still occurs eagerly.   

The logic could probably be simplified using `Map::merge` within the `for` loop and skipping the creation of `scalarResourceMap`.

Reply:
thanks for the tip on Java 8 👍